src/test/deploy/functions/services/authBlocking.spec.ts

```diff
@@ -0,0 +1,527 @@
+        },
+      };
+      const ep: backend.Endpoint = {
+        ...BASE_EP,
+        platform: "gcfv1",
+        uri: "somethingnew.url",
+        blockingTrigger: {
+          eventType: BEFORE_CREATE_EVENT,
+          accessToken: false,
+          idToken: true,
+          refreshToken: false,
+        },
+      };
+      getConfig.resolves(blockingConfig);
+      setConfig.resolves(newBlockingConfig);
+
+      await authBlocking.registerTrigger(ep, true);
+
+      expect(blockingConfig).to.deep.equal(newBlockingConfig);
+      expect(setConfig).to.have.been.calledWith("project", newBlockingConfig);
+    });
+
+    it("should register on an update to a beforeSignIn endpoint", async () => {
+      const blockingConfig: identityPlatform.BlockingFunctionsConfig = {
+        triggers: {
+          beforeCreate: {
+            functionUri: "beforecreate.url",
+          },
+          beforeSignIn: {
+            functionUri: "beforesignin.url",
+          },
+        },
+        forwardInboundCredentials: {
+          accessToken: true,
+        },
+      };
+      const newBlockingConfig: identityPlatform.BlockingFunctionsConfig = {
+        triggers: {
+          beforeCreate: {
+            functionUri: "beforecreate.url",
+          },
+          beforeSignIn: {
+            functionUri: "somethingnew.url",
+          },
+        },
+        forwardInboundCredentials: {
+          accessToken: true,
+        },
+      };
+      const ep: backend.Endpoint = {
+        ...BASE_EP,
+        platform: "gcfv1",
+        uri: "somethingnew.url",
+        blockingTrigger: {
+          eventType: BEFORE_SIGN_IN_EVENT,
+          accessToken: false,
+          idToken: true,
+          refreshToken: false,
+        },
+      };
+      getConfig.resolves(blockingConfig);
+      setConfig.resolves(newBlockingConfig);
+
+      await authBlocking.registerTrigger(ep, true);
+
+      expect(blockingConfig).to.deep.equal(newBlockingConfig);
+      expect(setConfig).to.have.been.calledWith("project", newBlockingConfig);
+    });
+
+    it("should do not set the config if the config is unchanged", async () => {
+      const blockingConfig: identityPlatform.BlockingFunctionsConfig = {
+        triggers: {
+          beforeCreate: {
+            functionUri: "somethingnew.url",
+          },
+          beforeSignIn: {
+            functionUri: "beforesignin.url",
+          },
+        },
+        forwardInboundCredentials: {
+          accessToken: true,
+        },
+      };
+      const ep: backend.Endpoint = {
+        ...BASE_EP,
+        platform: "gcfv1",
+        uri: "somethingnew.url",
+        blockingTrigger: {
+          eventType: BEFORE_CREATE_EVENT,
+          accessToken: true,
+        },
+      };
+      getConfig.resolves(blockingConfig);
+
+      await authBlocking.registerTrigger(ep, false);
+
+      expect(setConfig).to.not.have.been.called;
+    });
+  });
+
+  describe("unregisterAuthBlockingTriggerFromIdentityPlatform", () => {
```

Comment:
You can't use spaces here?

Reply:
Oops, leftover from a previous name, I'll change it to `unregisterTrigger`